lib/new_relic/agent/new_relic_service/json_marshaller.rb

```diff
@@ -42,8 +42,8 @@ def load(data)
             return nil
           end
 
-          return_value(::JSON.load(data))
-        rescue => e
+          return_value(::JSON.parse(data))
+        rescue JSON::ParserError => e
```

Comment:
Thank you for working on this! Just in case an unexpected error occurs within the `load` method, it would be great to continue rescuing all errors. 
```suggestion
        rescue => e
```